webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx

```diff
@@ -357,6 +361,27 @@ export const GlideTable: React.FC<GlideTableProps> = ({
             }, 5);
           },
         },
+        // Column is pinned if the index is inside of the frozen columns
+        col > pinnedColumnsCount + STATIC_COLUMNS.length - 1
+          ? {
+              key: 'pin',
+              label: 'Pin column',
+              onClick: () => {
+                const newSortableColumns = sortableColumnIds.filter((c) => c !== column.column);
+                newSortableColumns.splice(pinnedColumnsCount, 0, column.column);
+                setSortableColumnIds(newSortableColumns);
+                setPinnedColumnsCount(Math.min(pinnedColumnsCount + 1, sortableColumnIds.length));
```

Comment:
suggestion: This feels fragile/complicated to me. What if instead we handled pinning by tracking an array of `pinnedColumns` and defining columnIds like this:
```
const columnIds = useMemo(() => 
   [...STATIC_COLUMNS, ...pinnedColumns, ...sortableColumnIds.filter(c => !pinnedColumns.includes(c)], 
[pinnedColumns, sortableColumnIds]);
```

Reply:
i preferred keeping the count instead because it allowed for rearranging the pinned columns while also meaning we didn't need to ensure the pinned column array in sync with the sortable column array. That said, i don't think there's a way of doing that while also allowing the current behavior w/r/t the compare view, so i'm rewriting it to just have the list as you have here.

Reply:
Tooled around with this approach a bit and it's making the column moving code kinda hairy because a core assumption there is of the grid coordinates lining up with the sortable columns array. Note that moving the column is unavoidable anyway, because we want columns to appear in the leftmost position when unpinned.